pySDC/tests/test_projects/test_DAE/test_SemiImplicitDAE.py

```diff
@@ -0,0 +1,354 @@
+import pytest
+import numpy as np
+
+
+def getTestSetup(problem, sweeper, hook_class):
+    r"""
+    Returns the description for the tests.
+
+    Parameters
+    ----------
+    hook_class : list
+        Hook classes to log statistics such as errors.
+
+    Returns
+    -------
+    description : dict
+        Contains the parameters for one run.
+    controller_params : dict
+        Controller specific parameters.
+    """
+
+    level_params = {
+        'restol': 1e-13,
+    }
+
+    problem_params = {
+        'newton_tol': 1e-6,
+    }
+
+    step_params = {
+        'maxiter': 60,
+    }
+
+    controller_params = {
+        'logger_level': 30,
+        'hook_class': hook_class,
+    }
+
+    description = {
+        'problem_class': problem,
+        'problem_params': problem_params,
+        'sweeper_class': sweeper,
+        'level_params': level_params,
+        'step_params': step_params,
+    }
+    return description, controller_params
+
+
+@pytest.mark.base
+@pytest.mark.parametrize('initial_guess', ['spread', 'zero', 'random'])
+def testPredict(initial_guess):
+    r"""
+    In this test the predict function of the sweeper is tested.
+    """
+
+    from pySDC.projects.DAE.sweepers.SemiImplicitDAE import SemiImplicitDAE
+    from pySDC.projects.DAE.problems.DiscontinuousTestDAE import DiscontinuousTestDAE
+    from pySDC.projects.DAE.misc.DAEMesh import DAEMesh
+    from pySDC.core.Step import step
+
+    description, _ = getTestSetup(DiscontinuousTestDAE, SemiImplicitDAE, [])
+
+    sweeper_params = {
+        'quad_type': 'RADAU-RIGHT',
+        'num_nodes': 2,
+        'QI': 'IE',
+        'initial_guess': initial_guess,
+    }
+    description.update({'sweeper_params': sweeper_params})
+
+    level_params = description['level_params']
+    level_params.update({'dt': 0.1})
+    description.update({'level_params': level_params})
+
+    S = step(description=description)
+    L = S.levels[0]
+    P = L.prob
+
+    assert isinstance(L.sweep, SemiImplicitDAE), "Sweeper cannot instantiate an object of type SemiImplicitDAE!"
+
+    L.status.time = 1.0
+    L.u[0] = P.u_exact(L.time)
+
+    assert isinstance(L.f[0], type(None)), "Initial condition f0 is not of type None!"
```

Comment:
Why do you test this? This seems to be a design choice irrespective or your sweeper.

Reply:
True. I'll remove that.